python-checks/src/main/java/org/sonar/python/checks/PandasAddMergeParametersCheck.java

```diff
@@ -27,19 +27,57 @@
 import org.sonar.plugins.python.api.PythonSubscriptionCheck;
 import org.sonar.plugins.python.api.SubscriptionContext;
 import org.sonar.plugins.python.api.symbols.Symbol;
+import org.sonar.plugins.python.api.tree.Argument;
 import org.sonar.plugins.python.api.tree.CallExpression;
 import org.sonar.plugins.python.api.tree.Tree;
 import org.sonar.python.tree.TreeUtils;
 
 @Rule(key = "S6735")
 public class PandasAddMergeParametersCheck extends PythonSubscriptionCheck {
 
-  private static final List<String> messages = List.of(
-    "The '%s' parameter of the merge should be specified.",
-    "The '%s' and '%s' parameters of the merge should be specified.",
-    "The '%s', '%s' and '%s' parameters of the merge should be specified.");
+  enum Keywords {
+    HOW("how", 2, 1, 2),
+    ON("on", 1, 2, 3),
+    VALIDATE("validate", 6, 11, 12);
 
-  private static final Set<String> MERGE_METHODS = Set.of("pandas.merge", "pandas.core.reshape.merge.merge");
+    public String getKeyword() {
+      return keyword;
+    }
+
+    public int getJoinPosition() {
+      return joinPosition;
+    }
+
+    public int getDataFrameMergePosition() {
+      return dataFrameMergePosition;
+    }
+
+    public int getPandasMergePosition() {
+      return pandasMergePosition;
+    }
+
+    final String keyword;
+    final int joinPosition;
+    final int dataFrameMergePosition;
+    final int pandasMergePosition;
+
```

Comment:
_Nitpick:_ Related to my comments about making things clearer with naming, here I think the fields and their getters could benefit from names like `positionInJoin` as they represent the position of the argument in the join method and not the position of join itself. I personally think it is better to have a longer name that describes the intent clearly rather than shorter names

Reply:
I agree that this would be better. I changed it in 2669345097d5e46d6ace627f7900ffe63f46d29f.